src/ppspp_options.erl

```diff
@@ -81,44 +82,49 @@
 -spec unpack(binary()) -> {binary(), options()} | {error, atom()}.
 %% options is an orddict.
 unpack(Maybe_Options) ->
-    {Maybe_Messages, Options} = unpack(Maybe_Options, orddict:new()),
-    {Maybe_Messages, {options, Options}}.
+    {Maybe_Messages, Options} = unpack(Maybe_Options, #{}),
+    {Maybe_Messages, Options}.
+
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
--spec unpack(binary(), options_dict()) -> { binary(), options_dict()}.
+-spec unpack(binary(), options()) -> { binary(), options()}.
 unpack( <<?PPSPP_SUPPORTED_VERSION,
           Version:?BYTE,
-          Maybe_Options/binary >>, Options0) ->
-    Options = orddict:store(supported_version, Version, Options0),
-    unpack(Maybe_Options, Options);
+          Maybe_Options/binary >>, Options) ->
+    %% Options = orddict:store(supported_version, Version, Options0),
+    unpack(Maybe_Options, Options#{supported_version => Version});
+
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 unpack( <<?PPSPP_MINIMUM_VERSION,
           Version:?BYTE,
-          Maybe_Options/binary >>, Options0) ->
-    Options = orddict:store(minimum_version, Version, Options0),
-    unpack(Maybe_Options, Options);
+          Maybe_Options/binary >>, Options) ->
+    %% Options = orddict:store(minimum_version, Version, Options0),
+    unpack(Maybe_Options, Options#{minimum_version => Version});
+
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 unpack( <<?PPSPP_SWARM_ID_LENGTH,
           Length:?WORD,
           Swarm_ID:Length/binary,
-          Maybe_Options/binary >>, Options0) ->
-    Options = orddict:store(swarm_id, Swarm_ID, Options0),
-    unpack(Maybe_Options, Options);
+          Maybe_Options/binary >>, Options) ->
+    %% Options = orddict:store(swarm_id, Swarm_ID, Options0),
+    unpack(Maybe_Options, Options#{swarm_id => Swarm_ID});
+
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 unpack( <<?PPSPP_INTEGRITY_CHECK_METHOD,
           Maybe_Method:?BYTE,
-          Maybe_Options/binary >>, Options0) ->
+          Maybe_Options/binary >>, Options) ->
     Method = case Maybe_Method of
                  0 -> no_integrity_protection;
                  1 -> merkle_hash_tree;
                  2 -> sign_all;
                  3 -> unified_merkle_hash_tree
              end,
-    Options = orddict:store(content_integrity_check_method, Method, Options0),
-    unpack(Maybe_Options, Options);
+    %% Options = orddict:store(content_integrity_check_method, Method, Options0),
```

Comment:
Line 121 is too long:     %% Options = orddict:store(content_integrity_check_method, Method, Options0),.